Header header 
ProposalRay[] Ray_Path

